cpg-language-go/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/golang/GoLanguageFrontend.kt

```diff
@@ -205,11 +211,28 @@ class GoLanguageFrontend(language: Language<GoLanguageFrontend>, ctx: Translatio
             if (declaration is DeclarationSequence) {
                 declaration.declarations.forEach { scopeManager.addDeclaration(it) }
             } else {
-                scopeManager.addDeclaration(declaration)
+                // We need to be careful with method declarations. We need to put them in the
+                // respective name scope of the record and NOT on the global scope / namespace scope
+                // TODO: this is broken if we see the declaration of the method before the class :(
```

Comment:
Can this happen? If so, doesn't this indicate that it's more likely to be a job for a pass?

Reply:
See https://github.com/Fraunhofer-AISEC/cpg/issues/1561